org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/WorkspaceDiagnosticsHandler.java

```diff
@@ -170,6 +176,8 @@ public boolean visit(IResourceDelta delta) throws CoreException {
 				markers = Arrays.copyOf(javaMarkers, javaMarkers.length + taskMarkers.length);
 				System.arraycopy(taskMarkers, 0, markers, javaMarkers.length, taskMarkers.length);
 				document = JsonRpcHelpers.toDocument(cu.getBuffer());
+			} else if (handler != null) {
+				handler.triggerValidation(cu);
```

Comment:
Do you know why the 400ms delay on triggering validation from the underlying method ? I believe it was added by @testforstephen when the Syntax Server was introduced.

Reply:
It is added by @aeschli 
You may want to take a look at
- https://github.com/eclipse/eclipse.jdt.ls/pull/425
- https://github.com/eclipse/eclipse.jdt.ls/pull/989
- https://github.com/eclipse/eclipse.jdt.ls/issues/847
- https://github.com/eclipse/eclipse.jdt.ls/issues/1198

Reply:
Right, makes sense. We don't want to trigger validation on every keystroke.